Tasks/Gradle/CodeAnalysis/Common/CodeAnalysisResultPublisher.ts

```diff
@@ -144,6 +147,12 @@ export class CodeAnalysisResultPublisher {
             return tl.loc('codeAnalysisBuildSummaryLine_OneViolationOneFile', toolName);
         }
         if (violationCount === 0) {
+            // Tools produce an AnalysisResult regardless of whether they were enabled through the UI or not
+            // Therefore, only show "X did not find any violations" messages if the tool was enabled
+            if (!analysisResultsGroup[0].originatingTool.isEnabled()) {
```

Comment:
This is the core of the fix - if the violation count is 0 and the tool was not enabled, do not return a line.